src/packages/ganache/webpack/webpack.common.config.ts

```diff
@@ -34,7 +34,6 @@ const base: webpack.Configuration = {
   output: {
     filename: "ganache.min.js",
     library: "Ganache",
-    libraryExport: "default",
```

Comment:
Any thoughts on if this change has an effect elsewhere?

Reply:
The only change I  noticed is that it manages to export the named exports in addition to the default export. I think if this is set to `default`, it only exports the default export.

Reply:
If we want to be really safe, I can isolate this change to just `webpack.cli.config`

Reply:
Ah, ok.

Can you run some smoke tests locally to ensure we can still import and `require` both the `node` and `cli` bundles?

basically:

```javascript
// index.js
const ganache = require("{path-to-file}/ganache.min.js");
console.log(ganache) // should contain `provider` and `server` (`default` show up here, too)
```

```typescript
// index.ts, with no/default tsconfig.json (i.e., `esModuleInterop` is set to `false`)
import ganache from "{path-to-file}/ganache.min.js";
console.log(ganache) // should contain `provider` and `server`
```

Reply:
important Note: I have NOT checked if this is how it behaved prior to your changes, but it how I'd like it to behave. :-)

Reply:
Case 1: `esModuleInterop` is set to `true`

```
 node test.js
{ serverDefaults: [Getter],
  default:
   { server: [Function: server], provider: [Function: provider] } }
(node:11690) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
```

Case 2: `esModuleInterop` is set to `false`

I can't even do a `npm run tsc`

```


index.ts:1:8 - error TS1259: Module '"/home/heyse/development/ganache-core/src/packages/promise-queue/node_modules/emittery/index"' can only be default-imported using the 'esModuleInterop' flag

1 import Emittery from "emittery";
         ~~~~~~~~

  node_modules/emittery/index.d.ts:414:1
    414 export = Emittery;
        ~~~~~~~~~~~~~~~~~~
    This module is declared with using 'export =', and can only be used with a default import when using the 'esModuleInterop' flag.

```

Reply:
Note I had to export `serverDefaults` to get `cli` working

Reply:
Ah ok, so we might need to export `server` and `provider` in addition to the default; and probably export `serverDefaults` as part of `server` (maybe call it something like `server.defaults`). Are you up for making these changes?

Reply:
> Ah ok, so we might need to export `server` and `provider` in addition to the default; and probably export `serverDefaults` as part of `server` (maybe call it something like `server.defaults`). Are you up for making these changes?

Sure, but I don't really get why `serverDefaults` needs to be a part of `server`, as `cli` is expecting it to be a named export from `@ganache-core`

Reply:
I don't think it really makes sense to have it exposed to the public interface at the level it is now. I'd really prefer if it wasn't exposed to users at all, but that'd likely be a lot more work.

Reply:
I wonder if we can easily move `serverDefaults` from `packages/core` over to `packages/cli`; that way we won't need to expose it at all!

Reply:
> I wonder if we can easily move `serverDefaults` from `packages/core` over to `packages/cli`; that way we won't need to expose it at all!

I don't think this is possible as `core` uses it in `src/packages/core/src/options/index.ts`

Reply:
> important Note: I have NOT checked if this is how it behaved prior to your changes, but it's how I'd like it to behave, if possible. :-)

fyi this is how it is currently in `develop`

```
node test.js 
{ server: [Function: server], provider: [Function: provider] }
```

Reply:
```
node test.js 
{ serverDefaults: [Getter],
  server: [Function: server],
  provider: [Function: provider],
  default:
   { server: [Function: server], provider: [Function: provider] } }
```